### PR TITLE
VZ-2044: Update Coherence ES index

### PR DESCRIPTION
**Pull request by fredtibbitts**
This PR updates the Coherence ES index to match the work that GZ recently did for WLS. The index uses the pattern `oam-#{ENV['NAMESPACE']}-#{ENV['APP_CONF_NAME']}-#{ENV['COMPONENT_NAME']}` so that users can more easily search logs for their applications.

The work in the VerrazzanoCoherenceWorkload controller was simply to update the fluentd rules and also to copy the OAM app and component name labels into the Coherence CR that the controller writes out.

I also moved 3 fluentd container environment variables out of `wls.go` into `fluentd.go` as they will be needed by all workload types.

### CODE REVIEW

**Review by guoyongzhang — APPROVED** [review]
LGTM

File: application-operator/controllers/cohworkload/coherenceworkload_controller.go [hunk 1]
```diff
@@ -57,7 +62,7 @@ multiline_flush_interval 20s
   port "#{ENV['ELASTICSEARCH_PORT']}"
   user "#{ENV['ELASTICSEARCH_USER']}"
   password "#{ENV['ELASTICSEARCH_PASSWORD']}"
-  index_name "#{ENV['COH_CLUSTER_NAME']}"
+  index_name "oam-#{ENV['NAMESPACE']}-#{ENV['APP_CONF_NAME']}-#{ENV['COMPONENT_NAME']}"
```
Comment (kminder) [inline application-operator/controllers/cohworkload/coherenceworkload_controller.go:65]: I'm not a fan of exposing oam any more than necessary.  Could this be vz- or verrazzano- instead.  I'm aware that this probably means more work elsewhere too.

Comment (fredtibbitts) [reply application-operator/controllers/cohworkload/coherenceworkload_controller.go:65]: I was following what GZ did. @guoyongzhang can you comment on having "oam" in the index?

Comment (fredtibbitts) [reply application-operator/controllers/cohworkload/coherenceworkload_controller.go:65]: Also we should probably make the index a constant in fluentd.go so we all use the same pattern. I can make that update.

Comment (guoyongzhang) [reply application-operator/controllers/cohworkload/coherenceworkload_controller.go:65]: how aout app-....?

Comment (kminder) [reply application-operator/controllers/cohworkload/coherenceworkload_controller.go:65]: Someone should check with @gilbode for an official opinion.

Comment (fredtibbitts) [reply application-operator/controllers/cohworkload/coherenceworkload_controller.go:65]: I'm not sure the "oam-" adds value, maybe just remove it altogether? I agree with @kminder we should get sign-off on the index pattern.

Comment (markxnelson) [reply application-operator/controllers/cohworkload/coherenceworkload_controller.go:65]: i see no reason to include `oam` in this name - the fact that we use oam to deploy something is an implementation detail

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go [hunk 2]
```diff
@@ -106,6 +108,9 @@ func TestReconcileCreateCoherence(t *testing.T) {
 		DoAndReturn(func(ctx context.Context, u *unstructured.Unstructured, opts ...client.CreateOption) error {
 			assert.Equal("coherence.oracle.com/v1", u.GetAPIVersion())
 			assert.Equal("Coherence", u.GetKind())
+
+			// make sure the oam component and app name labels were copied
```
Comment (markxnelson) [inline application-operator/controllers/cohworkload/coherenceworkload_controller_test.go:112]: OAM is captitalized

**Review by markxnelson — CHANGES_REQUESTED** [review]

File: application-operator/controllers/cohworkload/coherenceworkload_controller_test.go (hunk 2 above)
Comment (fredtibbitts) [reply application-operator/controllers/cohworkload/coherenceworkload_controller_test.go]: Fixed.

File: application-operator/controllers/cohworkload/coherenceworkload_controller.go (hunk 1 above)
Comment (fredtibbitts) [reply application-operator/controllers/cohworkload/coherenceworkload_controller.go]: I removed `oam-` and moved the index pattern to a shared constant.

File: application-operator/controllers/loggingscope/fluentd.go [hunk 3]
```diff
@@ -30,6 +31,9 @@ const (
 	elasticSearchPwdField  = "ELASTICSEARCH_PASSWORD"
 )
 
+// ElasticSearchIndex defines the common index pattern
+const ElasticSearchIndex = "#{ENV['NAMESPACE']}-#{ENV['APP_CONF_NAME']}-#{ENV['COMPONENT_NAME']}"
```
Comment (gilbode) [inline application-operator/controllers/loggingscope/fluentd.go:35]: Is this the same pattern used for other component types?  I'd like this to be consistent.

Comment (fredtibbitts) [reply application-operator/controllers/loggingscope/fluentd.go:35]: Yes, I made them all use this constant.

**Review by markxnelson — APPROVED** [review]

**Pull request merged**